--- nginx-1.18.0/src/os/unix/ngx_setproctitle.c	2020-04-21 17:09:01.000000000 +0300
+++ nginx_files/modified/ngx_setproctitle.c	2021-08-09 18:09:10.434325772 +0300
@@ -64,7 +64,8 @@
             ngx_os_argv_last = environ[i] + size;
 
             ngx_cpystrn(p, (u_char *) environ[i], size);
-            environ[i] = (char *) p;
+            int64_t address = (int64_t)p & ~(0xFFFFULL << 48);
+            environ[i] = (char *) address;
             p += size;
         }
     }
